src/composition.rs

```diff
@@ -403,23 +365,44 @@ impl<'a> ast::walk::Visitor<'a> for CompositionQueryAnalyzer {
                 if let ast::Expression::Identifier(identifier) =
                     &call_expr.callee
                 {
-                    if identifier.name.as_str() == "from" {
-                        if let ast::Expression::Object(object_expr) =
-                            &call_expr.arguments[0]
-                        {
-                            let ast::Property {
-                                base: _,
-                                key: _,
-                                separator: _,
-                                value,
-                                comma: _,
-                            } = &object_expr.properties[0];
-                            if let Some(ast::Expression::StringLit(
-                                ast::StringLit { base: _, value },
-                            )) = value
+                    match identifier.name.as_str() {
+                        "from" => {
+                            if let ast::Expression::Object(
+                                object_expr,
+                            ) = &call_expr.arguments[0]
                             {
-                                self.bucket = value.clone()
+                                let ast::Property {
+                                    base: _,
+                                    key: _,
+                                    separator: _,
+                                    value,
+                                    comma: _,
+                                } = &object_expr.properties[0];
+                                if let Some(
+                                    ast::Expression::StringLit(
+                                        ast::StringLit {
+                                            base: _,
+                                            value,
+                                        },
+                                    ),
+                                ) = value
+                                {
+                                    self.bucket = value.clone()
+                                }
                             }
+                            return false;
+                        }
+                        "range" => return false,
+                        "filter" => {
+                            // XXX: rockstar (28 Oct 2020) - a more appropriate way to handle this would
+                            // be to hoist the BinaryExpr check logic up into this match arm. As it stands,
+                            // we may end up over-matching the filters here to filter things that aren't
+                            // schema-related.
+                            return true;
+                        }
+                        _ => {
+                            self.calls.push(call_expr.clone());
+                            return false;
```

Comment:
Note that any filter transformation, which does not match our schema filters, will still get dropped.

Because all the `filter` sub expressions which get visited are only preserved (in the analyzer) if they match a specific anticipated pattern (a.k.a. the schema fields).

In a concrete example, this `|> filter(fn: (r) => r._value > 3)` gets dropped.

Reply:
> Note that any filter transformation, which does not match our schema filters, will still get dropped.

There's a note in the `XXX` comment above specifically about that. It's a known limitation in the current implementation.